app/auth/web.py

```diff
@@ -66,7 +66,9 @@ def get_valid_token(headers):
         return None
 
     keycloak_oidc_client = current_app.store.get_oauth_client(redis_key)
-    return keycloak_oidc_client.access_token
+    if hasattr(keycloak_oidc_client, "access_token"):
+        return keycloak_oidc_client.access_token
+    return None
```

Comment:
Should we log an error here before returning `none`? I think getting here means that we receive a cookie or bearer token identifying a logged-in user, but that we don't have a valid gitlab, JH or KC token in the redis store for this user. This shouldn't really happen, and if it happens the client should be informed about it such that it can initiate a re-login flow for the specific backend. We'll need to refactor this further down the road, so for now just logging an error would seem appropriate to me.